src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { Todo } from '../../types/Todo';
+import { Status } from '../../types/Status';
+import { Filter } from '../Filter';
+
+interface Props {
+  todos: Todo[];
+  filterBy: Status;
+  onAllclick: () => void;
+  onActiveClick: () => void;
+  onCompletedclick: () => void;
+}
+
+export const Footer: React.FC<Props> = React.memo(({
+  todos,
+  filterBy,
+  onAllclick,
+  onActiveClick,
+  onCompletedclick,
+}) => {
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="TodosCounter">
+        {`${todos.filter(todo => !todo.completed).length} items left`}
```

Comment:
Filter operation is directly used within the return statement. For better performance and readability, consider performing this operation before the return statement and storing the result in a variable.